gen3utils/gitops/gitops_validator.py

```diff
@@ -0,0 +1,385 @@
+        homepage = index.get("homepageChartNodes")
+        if homepage:
+            for item in homepage:
+                checks = ["node", "name"]
+                error = check_required_fields("components.index.homepage", checks, item)
+                if error:
+                    return error
+
+    explorerconfig = gitops.get("explorerConfig")
+    configs = []
+    if not explorerconfig:
+        dataConfig = gitops.get("dataExplorerConfig")
+        fileConfig = gitops.get("fileExplorerConfig")
+        if not dataConfig:
+            return FieldSyntaxError("(data)explorerConfig")
+        configs.append(dataConfig)
+        if fileConfig:
+            configs.append(fileConfig)
+    else:
+        configs = explorerconfig
+
+    for exp_config in configs:
+        filters = exp_config.get("filters")
+        if not filters:
+            return FieldSyntaxError("explorerConfig.filters")
+
+        tabs = filters.get("tabs")
+        if not tabs:
+            return FieldSyntaxError("explorerConfig.filters")
+
+        for tab in tabs:
+            checks = ["title", "fields"]
+            error = check_required_fields("explorerConfig.filters.tab", checks, tab)
+            if error:
+                return error
+
+        guppy = exp_config.get("guppyConfig")
+        if not guppy:
+            return FieldSyntaxError("explorerConfig.guppyConfig")
+        if not guppy.get("dataType"):
+            return FieldSyntaxError("explorerConfig.guppyConfig.dataType")
+
+        manifest_mapping = guppy.get("manifestMapping")
+        buttons = exp_config.get("buttons")
+        val_mapping = False
+        for button in buttons:
+            if button.get("enabled") and button.get("type") == "manifest":
+                val_mapping = True
+        if manifest_mapping and val_mapping:
+            checks = [
+                "resourceIndexType",
+                "resourceIdField",
+                "referenceIdFieldInResourceIndex",
+                "referenceIdFieldInDataIndex",
+            ]
+            error = check_required_fields(
+                "explorerConfig.guppyConfig.manifestMapping", checks, manifest_mapping
+            )
+            if error:
+                return error
+
+    study_viewer = gitops.get("studyViewerConfig")
+    if study_viewer:
+        checks = ["dataType", "listItemConfig", "rowAccessor"]
+        for viewer in study_viewer:
+            error = check_required_fields("studyViewerConfig", checks, viewer)
+            if error:
+                return error
+    return None
+
+
+def check_required_fields(path, checks, field):
+    for check in checks:
+        if not field.get(check):
+            return FieldSyntaxError(f"{path}.{check}")
+    return None
+
+
+def check_field_value(path, checks, accepted_values, errors):
+    for check in checks:
+        if check not in accepted_values:
+            errors.append(FieldError("Invalid field {} in {}".format(check, path)))
```

Comment:
there are a few places you return "Invalid field", i would suggest something a bit more explicit like "Field xxx in yyy is not in the ETL mapping". Similarly, `FieldSyntaxError` could explain that the field is not found _in the dictionary_.

These tests will report errors to users who are not always familiar with all the moving pieces so we should provide clear messages to help them fix their config